Retry transient download failures in operator build process (ROX-12397)

Example `curl` failure:

```
$ make -C operator kuttl
make: Entering directory '/go/src/github.com/stackrox/stackrox/operator'
push_pin
+ mkdir -p bin
+ curl --fail --location --output /go/src/github.com/stackrox/stackrox/operator//bin/kubectl-kuttl-0.11.0-verbose-resource https://github.com/porridge/kuttl/releases/download/v0.11.0-verbose-resource/kubectl-kuttl_0.11.0-verbose-resource_linux_x86_64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

 72 43.8M   72 31.5M    0     0   138M      0 --:--:-- --:--:-- --:--:--  138M
curl: (56) OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 104
make: *** [Makefile:196: kuttl] Error 56
make: Leaving directory '/go/src/github.com/stackrox/stackrox/operator'
```

It would be great to try and add a shell retry loop around the `curl` [invocation](https://github.com/stackrox/stackrox/blob/dca280f22269930b1184d8a48c047577977c02ad/operator/Makefile#L204) to retry such transient issues.

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -201,9 +201,10 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 	[ -x $${1} ] || { \
 		set -euxo pipefail ;\
 		mkdir -p bin ;\
-		for i in {1..5}; do \
+		attempts=5 ;\
+		for i in $$(seq $$attempts); do \
 			curl --fail --location --output $${1} $${2} && break ;\
-			[ $$i -eq 5 ] && exit 1; \
+			[ $$i -eq $$attempts ] && exit 1; \
 		done ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\
 		chmod +x $${1} ;\

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -201,7 +201,7 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 	[ -x $${1} ] || { \
 		set -euxo pipefail ;\
 		mkdir -p bin ;\
-		curl --fail --retry 5 --location --output $${1} $${2} ;\
+		curl --fail --retry 5 --retry-all-errors --location --output $${1} $${2} ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\
 		chmod +x $${1} ;\
 	} \

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -204,8 +204,9 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 		for i in {1..5}; do \
 			set +e ;\
 			curl --fail --location --output $${1} $${2} ;\
-			[ $$? -eq 0 ] && break ;\
+			code=$$? ;\
 			set -e ;\
+			[ $$code -eq 0 ] && break ;\
 			[ $$i -eq 5 ] && exit 1; \
 		done ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -201,7 +201,7 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 	[ -x $${1} ] || { \
 		set -euxo pipefail ;\
 		mkdir -p bin ;\
-		curl --fail --location --output $${1} $${2} ;\
+		curl --fail --retry 5 --location --output $${1} $${2} ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\
 		chmod +x $${1} ;\
 	} \

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -201,7 +201,13 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 	[ -x $${1} ] || { \
 		set -euxo pipefail ;\
 		mkdir -p bin ;\
-		curl --fail --retry 5 --retry-all-errors --location --output $${1} $${2} ;\
+		for i in {1..5}; do \
+			set +e ;\
+			curl --fail --location --output $${1} $${2} ;\
+			[ $$? -eq 0 ] && break ;\
+			set -e ;\
+			[ $$i -eq 5 ] && exit 1; \
+		done ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\
 		chmod +x $${1} ;\
 	} \

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -202,11 +202,7 @@ GET_GITHUB_RELEASE_FN = get_github_release() { \
 		set -euxo pipefail ;\
 		mkdir -p bin ;\
 		for i in {1..5}; do \
-			set +e ;\
-			curl --fail --location --output $${1} $${2} ;\
-			code=$$? ;\
-			set -e ;\
-			[ $$code -eq 0 ] && break ;\
+			curl --fail --location --output $${1} $${2} && break ;\
 			[ $$i -eq 5 ] && exit 1; \
 		done ;\
 		[[ "$$(uname -s)" != "Darwin" ]] || xattr -c $${1} ;\

